packages/owlbot-bootstrapper/common-container/fetch-api-info.ts

```diff
@@ -0,0 +1,149 @@
+  FileNotFoundError,
+  RepositoryFileCache,
+} from '@google-automations/git-file-utils';
+
+/**
+ * Function that gets api information from a public DRIFT bucket
+ * @param apiId the unique API ID
+ * @param storageClient an instance of Google Cloud Storage
+ * @returns DriftApi info
+ */
+async function getDriftMetadata(
+  apiId: string,
+  storageClient: Storage
+): Promise<DriftApi> {
+  const bucket = 'devrel-prod-settings';
+  const file = 'apis.json';
+  const contents = await storageClient.bucket(bucket)?.file(file)?.download();
+  let apis;
+  if (contents) {
+    apis = JSON.parse(contents.toString()).apis;
+  }
+
+  return extractApiInfoFromJson(apis, apiId);
+}
+/**
+ * Function that extracts information for a given API from apis.json
+ *
+ * @param apis the apis.json file downloaded from drift as a JSON object
+ * @param apiId the unique identifier of the API
+ * @returns DriftAPI the fields as described in Drift
+ */
+function extractApiInfoFromJson(apis: DriftApi[], apiId: string): DriftApi {
+  const shortName = apiId.split('.')[apiId.split('.').length - 2];
+  for (const api of apis) {
+    if (api.api_shortname?.includes(shortName)) {
+      return api;
+    }
+  }
+
+  return {
+    api_shortname: shortName,
+    display_name: '',
+    docs_root_url: '',
+    launch_stage: 'LAUNCH_STAGE_UNSPECIFIED' as ReleaseLevel,
+    github_label: '',
+  };
+}
+
+/**
+ * Function that gets information from the service_config.yaml file in googleapis/googleapis
+ *
+ * @param apiId unique identifier fo API ID
+ * @param repositoryFileCache wrapper for getting repository contents from github (allows globbing)
+ * @returns a yaml object
+ */
+async function getApiProtoInformation(
+  apiId: string,
+  repositoryFileCache: RepositoryFileCache
+): Promise<ServiceConfigYaml | any> {
+  const path = apiId.toString().replace(/\./g, '/');
+
+  let yamlFile;
+  try {
+    yamlFile = await repositoryFileCache.getFileContents(
+      `${path}/.*?_v.*?.yaml$`,
+      'main'
+    );
+  } catch (e) {
+    if (e instanceof FileNotFoundError) {
+      console.log('service_config.yaml was not found');
+    } else {
+      // rethrow
+      throw e;
+    }
+  }
+
+  let parsedYaml;
+  if (yamlFile && yamlFile.parsedContent) {
+    parsedYaml = yaml.load(yamlFile.parsedContent);
+  }
+
+  if (!parsedYaml) {
+    throw new Error('Service config not valid yaml or undefined');
+  }
+
+  return parsedYaml;
+}
+
+function assignDriftValuesToServiceConfig(
+  serviceConfig: ServiceConfigYaml,
+  driftData: DriftApi
+): ServiceConfigYaml {
+  if (!serviceConfig?.api_short_name) {
+    serviceConfig.api_short_name = driftData.api_shortname;
+  }
+
+  if (!serviceConfig?.github_label) {
+    serviceConfig.github_label = driftData.github_label;
+  }
+
+  if (!serviceConfig?.documentation_uri) {
+    serviceConfig.documentation_uri = driftData.docs_root_url;
+  }
+
+  if (!serviceConfig?.launch_stage) {
+    serviceConfig.launch_stage = driftData.launch_stage;
+  }
```

Comment:
Consider adding logging so that we can inspect which fields are being taken from drift

Reply:
Debug logging.